bundle/config/mutator/initialize_urls.go

```diff
@@ -0,0 +1,62 @@
+package mutator
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/databricks/cli/bundle"
+	"github.com/databricks/cli/libs/diag"
+)
+
+type initializeUrls struct {
+	name string
+}
+
+// InitializeURLs makes sure the URL field of each resource is configured.
+// NOTE: since this depends on an extra API call, this mutator adds some extra
+// latency. As such, it should only be used when needed.
+// This URL field is used for the output of the 'bundle summary' CLI command.
+func InitializeURLs() bundle.Mutator {
+	return &initializeUrls{}
+}
+
+func (m *initializeUrls) Name() string {
+	return fmt.Sprintf("ConfigureURLs(%s)", m.name)
+}
+
+func (m *initializeUrls) Apply(ctx context.Context, b *bundle.Bundle) diag.Diagnostics {
+	workspaceId, err := b.WorkspaceClient().CurrentWorkspaceID(ctx)
+	orgId := strconv.FormatInt(workspaceId, 10)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	configureForOrgId(b, orgId)
+	return nil
+}
+
+func configureForOrgId(b *bundle.Bundle, orgId string) {
+	urlPrefix := b.Config.Workspace.Host
+	if !strings.HasSuffix(urlPrefix, "/") {
+		urlPrefix += "/"
+	}
+	urlSuffix := ""
+
+	// Add ?o=<workspace id> only if <workspace id> wasn't in the subdomain already.
+	// The ?o= is needed when vanity URLs / legacy workspace URLs are used.
+	// If it's not needed we prefer to leave it out since these URLs are rather
+	// long for most terminals.
```

Comment:
I believe you're talking about Azure hostnames here? If so, can you clarify in the comment where this holds.

Reply:
It doesn't really depend on Azure. Vanity URLs can occur for all clouds, as e.g. seen with some internal dogfooding/field workspaces. And modern Azure workspaces all encode the workspace id.

Reply:
It doesn't really depend on Azure. Vanity URLs can occur for all clouds, as e.g. seen with some internal dogfooding/field workspaces. And modern Azure workspaces all encode the workspace id.

Reply:
I don't understand. There are vanity URLs on all clouds. And we'll get many more than we have today. Vanity URLs would indeed typically not encode the workspace ID but that is cloud-independent.

Is there a different text for this comment that you would suggest?